engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -670,7 +680,51 @@ else if ( this.sort != null && projection != null ) {
 		);
 	}
 
+	private void validate(Sort sort, Collection<Class<?>> classes) {
+		SortField[] sortFields = sort.getSort();
+		for ( SortField sortField : sortFields ) {
+			String field = sortField.getField();
+			boolean indexedField = false;
+			for ( Class<?> clazz : classes ) {
+				IndexedTypeDescriptor indexedTypeDescriptor = extendedIntegrator.getIndexedTypeDescriptor( clazz );
+				FieldDescriptor fieldDescriptor = indexedTypeDescriptor.getIndexedField( field );
```

Comment:
Here you are using the public metadata API opposed to the internal one (as you do later). `FieldDescriptor` seems to be the right spot to add additional metadata. There is for example already a `NumericFieldSettingsDescriptor`, maybe a `SpatialFieldSettingsDescriptor` would make sense. This way you might be able to skip accessing the internal metadata alltogether (obviously the internal metadata would need to change as well, since the public one is built from it, just saying you might be able to just use one type of API in this particular usecase)